syntax = "proto3";

// make new paste
message newPasteArgs {
    int64 user_id = 1;
    string text = 2;
    string title = 3;
    string password = 4;
};

message newPasteResponce {
    string public_key = 1;
}

// get paste info
message getPasteInfoArgs {
    string public_key = 1;
}

message getPasteInfoResponce {
    string author = 1;
    string password = 2;
    string title = 3;
    string created_at = 4;
}

// get paste text
message getPasteTextArgs {
    string public_key = 1;
}

message getPasteTextResponce {
    string paste_text = 1;
}

// delete paste
message delPasteArgs {
    string public_key = 1;
}

message delPasteResponce {
    bool is_success = 1;
}

// update paste info
message updatePasteArgs {
    string public_key = 1;
    string new_password = 2;
    string new_title = 3;
}

message updatePasteResponce {
    bool is_success = 1;
}

// get last n pastes
message getLastePastesArgs {
    int64 user_id = 1;
    int32 count = 2;
}

message getLastePastesResponce {
    string last_pastes = 1;
}

service pastebinApi {
    rpc makeNewPaste(newPasteArgs) returns(newPasteResponce) {}
    rpc getPasteInfo(getPasteInfoArgs) returns(getPasteInfoResponce) {}
    rpc getPasteText(getPasteTextArgs) returns(getPasteTextResponce) {}
    rpc deletePaste(delPasteArgs) returns(delPasteResponce) {}
    rpc updatePaste(updatePasteArgs) returns(updatePasteResponce) {}
    rpc getLastPastes(getLastePastesArgs) returns(getLastePastesResponce) {}
}


